Credential Service — minor refactor

XVincentX opened the pull request:
Connects #361 

This pull request will change several things in the credential service. As I'll be working on this soon due to #361 , it's good to have a better base.

1. Reduce promise nesting (which is always bad)
2. DRY some functions and reuse them where possible
3. Throw errors instead of rejecting promises (an exception in a promise *IS* a rejection)

This is easier to review disabling whitespace diffs: https://github.com/ExpressGateway/express-gateway/pull/485/files?w=1

XVincentX commented:
@DrMegavolt Tests are passing — I'm missing permission to retry AppVeyor //cc @kevinswiber

DrMegavolt commented on test/services/auth.test.js:
```diff
@@ -77,102 +77,102 @@ describe('Auth tests', function () {
-            res.should.eql(true);
-            done();
-          });
+          done();
```
just remove the `done` callback as well as `catch`.

DrMegavolt commented on test/services/auth.test.js:
```diff
@@ -77,102 +77,102 @@ describe('Auth tests', function () {
+                  done();
+                });
+            })
+            .catch(function (err) {
```
same here, just return the promise and remove done

DrMegavolt commented on test/services/auth.test.js:
```diff
@@ -182,97 +182,97 @@ describe('Auth tests', function () {
+          should.exist(authResponse);
+          Object.keys(authResponse.token).sort().should.eql(expectedTokenProps.sort());
+          Object.keys(authResponse.consumer).sort().should.eql(expectedConsumerProps.sort());
+          delete authResponse.token.expiresAt;
```
can we use should.haveProperties ?

DrMegavolt commented on test/services/auth.test.js:
```diff
@@ -182,97 +182,97 @@ describe('Auth tests', function () {
+          authResponse.should.eql(false);
+          done();
+        })
+        .catch(function (err) {
```
rm done\catch

DrMegavolt commented on test/services/credential-service/credentials.test.js:
```diff
@@ -32,16 +32,16 @@ describe('Credential service tests', function () {
-        done();
-      });
+        .then(function (didSucceed) {
+          if (!didSucceed) {
```
not needed didSucceed is always true

DrMegavolt commented on test/services/credential-service/credentials.test.js:
```diff
@@ -107,66 +107,62 @@ describe('Credential service tests', function () {
+          newCredential.isActive.should.eql(true);
+          done();
+        })
+        .catch(done);
```
just return promise

DrMegavolt commented on test/services/credential-service/credentials.test.js:
```diff
@@ -231,40 +227,36 @@ describe('Credential service tests', function () {
-        should.exist(res);
-        credentialService.removeCredential(user.username, 'oauth2')
-        .then(deleted => {
-          deleted.should.eql(true);
```
you removed this check and check for  `resAfterDelete`, why ? 
what if some code relies on it returning true\false\undefined ?

XVincentX replied:
Good point, better to restore those.

DrMegavolt commented on test/services/credential-service/credentials.test.js:
```diff
@@ -289,16 +281,16 @@ describe('Credential service tests', function () {
       };
 
       db.flushdbAsync()
```
just return `db.flushdbAsync()` , then catch is redundant

XVincentX commented:
@DrMegavolt I've addressed all your comments and went even further. I looked over the two test files and I've reduced the promise nesting to the bare minimum, deleted all old `done` callbacks as well as using `should` where possible (to check rejection, for example).

Please have yet another look. Once this gets merged, we can move on with the user id/names PR.
